### PR TITLE
Display related links in taxonomy sidebar

### PR DESCRIPTION
**Pull request by alecgibson**
This change updates the taxonomy sidebar to accept a list of related content items under each taxon, and updates the styling to match the latest designs.

This is a **breaking change** which will align the API to the slimmed-down version in [this update to `govuk_navigation_helpers`](https://github.com/alphagov/govuk_navigation_helpers/pull/44). Since taxonomy display is still in alpha, and this is a novel component being used only by the navigation team this breakage is assumed to be acceptable.

# Screenshots

## Before

sidebar-before

## After

sidebar-after

# Trello

https://trello.com/c/EJXC60k1/429-add-more-like-this-results-to-side-navigation-in-frontend-apps

### CODE REVIEW

File: test/govuk_component/taxonomy_sidebar_test.rb [hunk 1]
```diff
@@ -13,53 +13,101 @@ def component_name
 
   test "renders a related items block" do
```
Comment (mobaig) [inline test/govuk_component/taxonomy_sidebar_test.rb]: This was actually my doing from a previous PR, but could you update this test description to say 'taxonomy sidebar' instead?

File: test/govuk_component/taxonomy_sidebar_test.rb [hunk 2]
```diff
@@ -13,53 +13,101 @@ def component_name
...
+          related_content: [
+            {
+              title: "Related link 1",
+              link: "/related-link-1",
+            },
             {
-              title: "Item title",
-              url: "/item",
-              description: "An item",
+              title: "Related link 2",
```
Comment (mobaig) [inline test/govuk_component/taxonomy_sidebar_test.rb]: Can we remove data for Related link 2 and 3 if we aren't making assertions about them below? Unnecessary test setup.

Comment (alecgibson) [inline test/govuk_component/taxonomy_sidebar_test.rb]: Sorry stupid copy-paste error.

File: test/govuk_component/taxonomy_sidebar_test.rb (hunk 1 above)
Comment (alecgibson) [inline test/govuk_component/taxonomy_sidebar_test.rb]: Will do

**Comment by alecgibson** [comment]
@mobaig changes pushed - can you please re-review?

**Comment by fofr** [comment]
Are those list link styles intentional?

**Comment by alecgibson** [comment]
@fofr you mean the bullets out to the left? Those are.

**Comment by fofr** [comment]
I don't know how the front-end community should review this and other recent fronted PRs.

We're having trouble knowing:
* What to treat as production ready
* What is something purely for prototyping, user research, integration only or AB testing on live
* What users will see these changes
* How to test/view the PR itself (ie in this example, the component being used in context)
* How to consider these changes inline with our other patterns
* How to review the design elements
* How people in the future will know to remove code that's not live

Neither the PR or the story give this information.

**Comment by tijmenb** [comment]
As discussed offline, agree it's probably hard to follow the process from outside of the team. We'll have a think and discuss tomorrow IRL!
